packages/firestore/test/integration/api/query.test.ts

```diff
@@ -2061,34 +2062,113 @@ apiDescribe('Queries', (persistence: boolean) => {
-  // sending existence filter in response: b/270731363. Remove the condition  
-  // here once the bug is resolved.
+  // TODO(b/270731363): Re-enable this test to run against the Firestore
+  // emulator once the bug where an existence filter fails to be sent when a
+  // query is resumed is fixed.
   // eslint-disable-next-line no-restricted-properties
-  (USE_EMULATOR ? it.skip : it)(
-    'resuming a query should remove deleted documents indicated by existence filter',
-    () => {
+  (USE_EMULATOR || !persistence ? it.skip : it)(
+    'resuming a query should use bloom filter to avoid full requery',
+    async () => {
+      // Create 100 documents in a new collection.
       const testDocs: { [key: string]: object } = {};
       for (let i = 1; i <= 100; i++) {
         testDocs['doc' + i] = { key: i };
       }
-      return withTestCollection(persistence, testDocs, async (coll, db) => {
-        const snapshot1 = await getDocs(coll);
-        expect(snapshot1.size).to.equal(100);
-        // Delete 50 docs in transaction so that it doesn't affect local cache.
-        await runTransaction(db, async txn => {
-          for (let i = 1; i <= 50; i++) {
-            txn.delete(doc(coll, 'doc' + i));
+
+      let attemptNumber = 0;
+      while (true) {
+        attemptNumber++;
+        const iterationResult = await withTestCollection<'retry' | 'passed'>(
+          persistence,
+          testDocs,
+          async (coll, db) => {
+            // Run a query to populate the local cache with the 100 documents
+            // and a resume token.
+            const snapshot1 = await getDocs(coll);
+            expect(snapshot1.size, 'snapshot1.size').to.equal(100);
+
+            // Delete 50 of the 100 documents. Do this in a transaction, rather
+            // than deleteDoc(), to avoid affecting the local cache.
+            await runTransaction(db, async txn => {
+              for (let i = 1; i <= 50; i++) {
+                txn.delete(doc(coll, 'doc' + i));
+              }
+            });
+
+            // Wait for 10 seconds, during which Watch will stop tracking the
+            // query and will send an existence filter rather than "delete"
+            // events when the query is resumed.
+            await new Promise(resolve => setTimeout(resolve, 10000));
+
+            // Resume the query and expect to get a snapshot with the 50
+            // remaining documents. Use some internal testing hooks to "capture"
+            // the existence filter mismatches to later verify that Watch sent a
+            // bloom filter, and it was used to avert a full requery.
+            const existenceFilterMismatches =
+              await captureExistenceFilterMismatches(async () => {
+                const snapshot2 = await getDocs(coll);
+                expect(snapshot2.size, 'snapshot2.size').to.equal(50);
+              });
+
+            // Verify that upon resuming the query that Watch sent an existence
```

Comment:
Update this comment once the "if" statement is changed to check for `persistence` as well.